gqcp/include/Processing/DensityMatrices/SpinResolvedOneDM.hpp

```diff
@@ -128,14 +121,25 @@ class SpinResolvedOneDM {
     }
 
     /**
-     *  @return the spin resolved density matrix in the underrlying scalar basis.
+     *  @param T_a          transformation matrix for the alpha component of the spin resolved 1-DM
+     *  @param T_b          transformation matrix for the beta component of the spin resolved 1-DM
+     * 
+     *  @return the transformed spin resolved density matrix, with each component transformed seperately t a different basis.
      */
-    SpinResolvedOneDM<Scalar> transformToScalarBasis(const TransformationMatrix<double>& C_a, const TransformationMatrix<double>& C_b) const {
-        OneDM<Scalar> D_AO_a = C_a.conjugate() * this->alpha() * C_a.transpose();
-        OneDM<Scalar> D_AO_b = C_b.conjugate() * this->beta() * C_b.transpose();
-        SpinResolvedOneDM<Scalar> D_AO {D_AO_a, D_AO_b};
+    SpinResolvedOneDM<Scalar> transform(const TransformationMatrix<double>& T_a, const TransformationMatrix<double>& T_b) const {
+        OneDM<Scalar> D_a_transformed = T_a.conjugate() * this->alpha() * T_a.transpose();
+        OneDM<Scalar> D_b_transformed = T_b.conjugate() * this->beta() * T_b.transpose();
 
-        return D_AO;
+        return SpinResolvedOneDM<Scalar> {D_a_transformed, D_b_transformed};
+    }
+
+    /**
+     *  @param T          transformation matrix for the alpha and beta component of the spin resolved 1-DM
+     * 
+     *  @return the transformed spin resolved density matrix, with each component transformed to the same basis.
+     */
+    SpinResolvedOneDM<Scalar> transform(const TransformationMatrix<double>& T) const {
+        return this->transform(T, T);
     }
```

Comment:
Thanks for implementing these methods!

Since you're providing two `TransformationMatrix` arguments, I would propose that we can take the spin-resolved types even further. Can you implement the (generic) type `SpinResolvedTransformationMatrix`, which effectively encapsulates `T_alpha` and `T_beta`? I think the extra encapsulation will be useful in later code! As for the public methods on `SpinResolvedTransformationMatrix`:
- `alpha()`
- `beta()`
- `transformed()`
- `transform()`

These are quite similar to `TransformationMatrix` and other spin-resolved types. Spin-resolved types should then also obtain the API to be transformed with a `SpinResolvedTransformationMatrix`.

Reply:
I have made an issue for this, to which I assigned myself. That way, I can tackle this in a separate PR.